### PR TITLE
Adding support for Arduino Nano RP2040 Connect

### PR DESCRIPTION
**Pull request by BlitzCityDIY**
Adding CircuitPython support for the Arduino Nano RP2040 Connect. Have tested all digital and analog I/O, I2C, onboard LSM6DSOX. Uses the AT25SF128A external flash.

### CODE REVIEW

File: ports/raspberrypi/boards/arduino_nano_rp2040_connect/mpconfigboard.mk.save [hunk 1]
```diff
@@ -0,0 +1,11 @@
+SB_VID = 0x239A
```
Comment (ladyada) [inline ports/raspberrypi/boards/arduino_nano_rp2040_connect/mpconfigboard.mk.save:1]: delete this file

File: ports/raspberrypi/boards/arduino_nano_rp2040_connect/mpconfigboard.mk [hunk 2]
```diff
@@ -0,0 +1,11 @@
+USB_VID = 0x2E8A
+USB_PID = 0x0003
```
Comment (tannewt) [inline ports/raspberrypi/boards/arduino_nano_rp2040_connect/mpconfigboard.mk:2]: Where did this PID come from?

Comment (BlitzCityDIY) [reply ports/raspberrypi/boards/arduino_nano_rp2040_connect/mpconfigboard.mk:2]: I looked it up under Windows device manager

**Comment by anecdata** [comment]
Will this work with ESP32SPI for wifi?

**Comment by ladyada** [comment]
it... "should" :) ill try it out when i get one

**Review by tannewt — CHANGES_REQUESTED** [review]
Looks like you need to run pre-commit locally to fix formatting issues. Instructions for pre-commit are here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code

**Comment by BlitzCityDIY** [comment]
ohh apologies, I missed the step for manually running. will do that now

**Comment by BlitzCityDIY** [comment]
@tannewt is it okay if I close this PR and submit again? I think I set stuff up wrong on my desktop for github when I was forking so it's giving me some problems when I'm trying to commit the updated pre-commit files

**Comment by dhalbert** [comment]
No problem closing this and opening a new one. The numbers are free.

**Comment by BlitzCityDIY** [comment]
Okay, thanks! Closing to re-submit with the pre-commit run.

**Comment by ladyada** [comment]
https://circuitpython.org/board/arduino_nano_rp2040_connect/ !

**Pull request closed without merging**